doc/adr/index.rst

```diff
@@ -1,10 +1,10 @@
-Architectural Decision Records
-==============================
+Architectural decision records
```

Comment:
Did you mean to change this file?

Reply:
I did mean to change it. Just cleaning up some inconsistent word usage and aligning with IOG style guide in terms of not using initial caps for all words in headings.